src/qibolab/instruments/qm/config/config.py

```diff
@@ -0,0 +1,125 @@
+from dataclasses import dataclass, field
+from typing import Union
+
+from qibolab.components.configs import IqConfig, OscillatorConfig
+from qibolab.pulses import Pulse
+
+from ..components import OpxOutputConfig, QmAcquisitionConfig, QmChannel
+from .devices import *
+from .elements import *
+from .pulses import *
+
+__all__ = ["QmConfig"]
+
+
+@dataclass
+class QmConfig:
+    """Configuration for communicating with the ``QuantumMachinesManager``.
+
+    Contains nested ``dataclass`` objects and is serialized using ``asdict``
+    to be sent to the instrument.
+    """
+
+    version: int = 1
+    controllers: dict[str, Controller] = field(default_factory=dict)
+    octaves: dict[str, Octave] = field(default_factory=dict)
+    elements: dict[str, Element] = field(default_factory=dict)
+    pulses: dict[str, Union[QmPulse, QmAcquisition]] = field(default_factory=dict)
+    waveforms: dict[str, Waveform] = field(default_factory=dict)
+    digital_waveforms: dict = field(
+        default_factory=lambda: {"ON": {"samples": [(1, 0)]}}
```

Comment:
As far as I recall, this is a peculiarity of QM and needs to be registered for things to work. I believe during the initial installation they suggested to use it without much explanation why (or I don't remember the explation). I am not sure if this is still the case for the latest qm-qua versions, however I believe we now need this anyway because we are using digital triggers to allow the Octave's LO signal to pass when needed.

Reply:
Ok, that's a bit weird, but if that's the case, there is not much we can do.

The only possible improvement would be to move the dictionary as a top-level constant, and document. But the only sensible piece of information to add in these docs is the part about the Octave, and the claim "you may always need this in any case".

Reply:
4df6925e98093831c7242c27fcd44087c9ac6cd1